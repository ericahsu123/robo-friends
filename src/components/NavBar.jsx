import React from 'react';
import Heading from './Heading';
import SearchBox from './SearchBox';

const NavBar = () => {
  return (
    <div className="ui top fixed huge stackable menu">
      <Heading />
      <SearchBox />
    </div>
  );
};

export default NavBar;
