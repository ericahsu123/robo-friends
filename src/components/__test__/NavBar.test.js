import React from 'react';
// We're using our own custom render function and not RTL's render
// our custom utils also re-export everything from RTL
// so we can import fireEvent and screen here as well
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import NavBar from '../NavBar';

afterEach(cleanup);

test('<NavBar/> component matches snapshot', () => {
  const mockStore = configureStore([]);
  const store = mockStore({});
  const component = renderer.create(
    <Provider store={store}>
      <NavBar />
    </Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
