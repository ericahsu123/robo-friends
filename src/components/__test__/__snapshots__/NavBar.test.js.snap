// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> component matches snapshot 1`] = `
<div
  className="ui top fixed huge stackable menu"
>
  <div
    className="item"
  >
    <h1>
      RoboFriends
    </h1>
  </div>
  <div
    className="right item"
  >
    <div
      className="ui icon input"
    >
      <input
        aria-label="Search Robots"
        data-testid="search-input"
        onChange={[Function]}
        placeholder="Search robots..."
        type="text"
      />
      <i
        className="search icon"
      />
    </div>
  </div>
</div>
`;
